lib/saluki-app/src/memory.rs

```diff
@@ -228,7 +228,7 @@ pub async fn initialize_allocator_telemetry() -> Result<(), GenericError> {
     // We can't enforce, at compile-time, that the tracking allocator must be installed if a caller is trying to
     // initialize the allocator's reporting infrastructure... but we can at least warn them if we detect it's not
     // installed here at runtime.
-    if AllocationGroupRegistry::allocator_installed() {
+    if !AllocationGroupRegistry::allocator_installed() {
```

Comment:
This is entirely unrelated, but is a simple fix for a logic bug that leads to us always emitting a warning that the tracking allocator isn't installed when, in fact, it is.